build.gradle

```diff
@@ -2311,6 +2311,12 @@ project(':tools') {
       exclude group: 'junit', module: 'junit'
     }
     testImplementation libs.reload4j
+    testImplementation libs.apachedsCoreApi
```

Comment:
why we needs those changes?

Reply:
any feedback?

Reply:
Hi @chia7712, if we don't add these dependencies here, some imported packages in `MiniKdc.java` can't be found.

Reply:
Could you please share the error message and command used to reproduce error?

Reply:
I reverted the change. Let's take a look for CI result. Thanks for the review.

Reply:
Hi @chia7712, without `build.gradle` change, we run command `./gradlew :tools:test --tests "org.apache.kafka.tools.consumer.group.SaslClientsWithInvalidCredentialsTest"` and get error:

```
java.lang.NoClassDefFoundError: org/apache/directory/server/core/api/DirectoryService
	at kafka.api.SaslSetup.$init$(SaslSetup.scala:60)
	at kafka.api.AbstractSaslTest.<init>(AbstractSaslTest.scala:20)
	at org.apache.kafka.tools.consumer.group.SaslClientsWithInvalidCredentialsTest.<init>(SaslClientsWithInvalidCredentialsTest.java:52)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.ClassNotFoundException: org.apache.directory.server.core.api.DirectoryService
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
```

Reply:
That is interesting. `tools` module SHOULD imports those dependencies for sasl-related tests, but not sure why they were not failed before ...